src/backend/base/langflow/components/vectorstores/astradb.py

```diff
@@ -294,8 +314,9 @@ def map_cloud_providers(cls):
     def get_vectorize_providers(cls, token: str, environment: str | None = None, api_endpoint: str | None = None):
         try:
             # Get the admin object
-            admin = AstraDBAdmin(token=token, environment=environment)
-            db_admin = admin.get_database_admin(api_endpoint=api_endpoint)
+            client = DataAPIClient(environment=environment)
+            admin_client = client.get_admin()
+            db_admin = admin_client.get_database_admin(api_endpoint, token=token)
 
             # Get the list of embedding providers
             embedding_providers = db_admin.find_embedding_providers().as_dict()
```

Comment:
Just a minor suggestion FYI (by no means urgent): if you do not do `.as_dict()` you can navigate the response all the same, but accessing named attributes of classes instead of dictionary keys. If you do typed code that may help with correctness.

Reply:
Fixed :) thanks!! much cleaner